### PR TITLE
Possibly fixes taser not being consumed in ED construction.

### LINKED ISSUE #3156: Creating an ED did not consume/place the taser in the ED
##### Revision that this bug was encountered on (REQUIRED!! Type 'show-server-revision' ingame to find out - note that 'current revision' isn't helpful!): #####
2014-03-16
d189abcbae68d37468db7f257b71def30ad748d7

##### Which server were you playing on? (REQUIRED!! e.g. TG-Sibyl1, TG-Sibyl2, Badgerstation, Facepunch, Pinkstation, /VG/... etc.) #####
tgstation - Sibyl 

###### What did you expect to happen? ######
When the roboticist was constructing the ED-209, a taser would be consumed and/or placed into the ED

###### What happened instead? ######
The roboticist was confused about what step of construction he was on, when infact the taser was actually placed inside the ED (probably referenced but not actually deleted or placed into the ED). Basically the construction step for the taser worked and the taser was not consumed.

###### Why is this bad/What are the consequences? ######
Probably pretty terrible, the roboticist could stash away a taser and show security the ED, which would mean the roboticist gets a taser without any suspicion at all

##### Steps to reproduce the problem (remember to be specific! Those little details can save us hours of work!): #####
--NOT A REPRODUCTION, I AM SAYING WHAT I CAN REMEMBER OF THE SITUATION--
1. The roboticist has a frame with legs already ready, I was in the lab waiting for him, he was running around either trying to find me or something, anyway. He had the frame in his backpack and he takes it out
2. I drop the taser armor and helmet from backpack. He places the armor and helmet on the ed, and  welds it (I think he tried to weld it)
3. I THINK, he then places the taser into the ED, but it doesnt drop or anything, he gets confused on what step he is on, attempts to put wire coils and looks at the wiki.
4. Somehow works, we test the ED by setting the roboticist to arrest and works as intended (which sounds about right, because the taser was probably referenced, or bool ticked however bots even work, maybe they dont even care)
5. Taser also worked fine

##### Possibly related stuff (which gamemode was it? What were you doing at the time? Was anything else out of the ordinary happening?): #####
It was changeling round, doesnt seem to be related.

##### Anything else? #####
I wasnt the roboticist, nor do I remember who was, but the taser was DEFINITELY not consumed by the construction process, I really haven't tried to reproduce the problem But it does exist, I am a witness to the event.
Also a note of reproduction, is that this doesnt always occur, doing it normally seems to be fine. If you want to try and figure out how, you should try breaking "the rules".

### PR DESCRIPTION
**Pull request by Firecage**
This possibly fixes the rare issue with a taser not being consumed in the ED209's construction, though I cannot be 100% sure since this is a rare issue, almost unreproducable. I did test this several times and could not reproduce it. It MAY have been due to the buildstep being furthered before the taser was deleted, unlike the other steps in the ED construction where it was deleted before the buildstep was furthered, like with the helmet. I also made stape 6, just before the tasers step, be build_step++ instead of build_step = 7, like the rest of the construction code for the ED209 is.

Possibly Fixes #3156

### CODE REVIEW

**Comment by phil235** [comment]
I don't see how any of these changes would fix that bug. The only reason I can think of for the item to not be deleted would be if user.drop_item() runtimes.

File: code/game/machinery/bots/ed209bot.dm [hunk 1]
```diff
@@ -649,12 +649,12 @@ Auto Patrol[]"},
 					name = "taser ED-209 assembly"
 				else
 					return
+			user.drop_item()
+			qdel(W)
```
Comment (RemieRichards) [inline code/game/machinery/bots/ed209bot.dm]: Don't put this here.
it's needed further down.

```user << "<span class='notice'>You add [W] to [src].</span>"```

**Comment by Firecage** [comment]
@phil235 Well, the taser is the only item reported to have caused this issue during ED construction(twice), and this is the only thing I could see which had a difference to the previous steps.

**Comment by Firecage** [comment]
Eh, well. Phil is probably right. This prob wouldn't do shit to fix it.

**Pull request closed without merging**